hearpreprocess/pipeline.py

```diff
@@ -459,6 +459,24 @@ def split_train_test_val(self, metadata: pd.DataFrame):
         metadata.loc[metadata["split_key"].isin(test_split_keys), "split"] = "test"
         return metadata
 
+    def trim_event_metadata(self, metadata: pd.DataFrame, duration: float):
+        # Since the duration in the task config is in seconds convert to milliseconds
+        duration_ms = duration * 1000.0
+        assert "start" in metadata.columns
+        assert "end" in metadata.columns
+        trimmed_metadata = metadata.loc[lambda df: df["end"] < duration_ms]
```

Comment:
No wrong. You drop it if the start >= duration_ms.
If end >= duration_ms you change the end to duration_ms.